packages/dds/tree/src/feature-libraries/chunked-forest/codec/chunkDecoding.ts

```diff
@@ -217,13 +227,56 @@ export class NestedArrayDecoder implements ChunkDecoder {
  */
 export class InlineArrayDecoder implements ChunkDecoder {
 	public constructor(private readonly shape: EncodedInlineArray) {}
-	public decode(decoders: readonly ChunkDecoder[], stream: StreamCursor): TreeChunk {
+	public decode(
+		decoders: readonly ChunkDecoder[],
+		stream: StreamCursor,
+		getIncrementalFieldBatch?: (fieldKey: string) => EncodedFieldBatch,
+	): TreeChunk {
 		const length = this.shape.length;
 		const decoder = decoders[this.shape.shape] ?? oob();
 		const chunks: TreeChunk[] = [];
 		for (let index = 0; index < length; index++) {
-			chunks.push(decoder.decode(decoders, stream));
+			chunks.push(decoder.decode(decoders, stream, getIncrementalFieldBatch));
+		}
+		return aggregateChunks(chunks);
+	}
+}
+
+/**
+ * Decoder for {@link EncodedIncrementalShape}s.
+ */
+export class IncrementalFieldDecoder implements ChunkDecoder {
+	public constructor(
+		private readonly shape: EncodedIncrementalShape,
```

Comment:
this property (and constructor parameter), which I believe is always just a 0, seems and could be removed.